[🐛 Bug]: noVNC session not showing anything

### What happened?

After starting the standalone or hub docker container and trying to debug using noVNC it shows nothing.
With Chrome Version 108, it was working as usual, but with versions 120, 121, and 122 it seems not to work anymore when trying to inspect the session. 

![no_VNC_session](https://github.com/SeleniumHQ/docker-selenium/assets/11818804/07271060-9c60-4a1c-96d8-13940f1be55f)


### Command used to start Selenium Grid with Docker (or Kubernetes)

```shell
docker run -d -p 4444:4444 -p 7900:7900 --shm-size="2g" selenium/standalone-chrome:4.18.1-20240224
```


### Relevant log output

```shell
2024-02-29 09:41:15 2024-02-29 07:41:15,607 INFO Included extra file "/etc/supervisor/conf.d/selenium.conf" during parsing
2024-02-29 09:41:15 2024-02-29 07:41:15,609 INFO RPC interface 'supervisor' initialized
2024-02-29 09:41:15 2024-02-29 07:41:15,610 CRIT Server 'unix_http_server' running without any HTTP authentication checking
2024-02-29 09:41:15 2024-02-29 07:41:15,610 INFO supervisord started with pid 24
2024-02-29 09:41:16 2024-02-29 07:41:16,618 INFO spawned: 'xvfb' with pid 31
2024-02-29 09:41:16 2024-02-29 07:41:16,621 INFO spawned: 'vnc' with pid 32
2024-02-29 09:41:16 2024-02-29 07:41:16,623 INFO spawned: 'novnc' with pid 33
2024-02-29 09:41:16 2024-02-29 07:41:16,626 INFO spawned: 'selenium-standalone' with pid 34
2024-02-29 09:41:16 2024-02-29 07:41:16,735 INFO success: xvfb entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2024-02-29 09:41:16 2024-02-29 07:41:16,736 INFO success: vnc entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2024-02-29 09:41:16 2024-02-29 07:41:16,736 INFO success: novnc entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2024-02-29 09:41:16 2024-02-29 07:41:16,736 INFO success: selenium-standalone entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2024-02-29 09:41:16 2024-02-29 07:41:16,736 INFO exited: novnc (terminated by SIGTRAP; not expected)
2024-02-29 09:41:16 Appending Selenium options: --heartbeat-period 30
2024-02-29 09:41:16 Appending Selenium options: --reject-unsupported-caps false
2024-02-29 09:41:16 Setting up SE_NODE_GRID_URL...
2024-02-29 09:41:16 Selenium Grid Standalone configuration: 
2024-02-29 09:41:16 [network]
2024-02-29 09:41:16 relax-checks = true
2024-02-29 09:41:16 
2024-02-29 09:41:16 [node]
2024-02-29 09:41:16 session-timeout = "300"
2024-02-29 09:41:16 override-max-sessions = false
2024-02-29 09:41:16 detect-drivers = false
2024-02-29 09:41:16 drain-after-session-count = 0
2024-02-29 09:41:16 max-sessions = 1
2024-02-29 09:41:16 
2024-02-29 09:41:16 [[node.driver-configuration]]
2024-02-29 09:41:16 display-name = "chrome"
2024-02-29 09:41:16 stereotype = '{"browserName": "chrome", "browserVersion": "122.0", "platformName": "Linux", "goog:chromeOptions": {"binary": "/usr/bin/google-chrome"}}'
2024-02-29 09:41:16 max-sessions = 1
2024-02-29 09:41:16 
2024-02-29 09:41:16 Starting Selenium Grid Standalone...
2024-02-29 09:41:16 Tracing is disabled
2024-02-29 09:41:17 07:41:17.666 INFO [LoggingOptions.configureLogEncoding] - Using the system default encoding
2024-02-29 09:41:17 07:41:17.674 INFO [OpenTelemetryTracer.createTracer] - Using OpenTelemetry for tracing
2024-02-29 09:41:18 07:41:18.371 INFO [NodeOptions.getSessionFactories] - Detected 3 available processors
2024-02-29 09:41:18 07:41:18.429 INFO [NodeOptions.report] - Adding chrome for {"browserName": "chrome","browserVersion": "122.0","goog:chromeOptions": {"binary": "\u002fusr\u002fbin\u002fgoogle-chrome"},"platformName": "linux","se:noVncPort": 7900,"se:vncEnabled": true} 1 times
2024-02-29 09:41:18 07:41:18.451 INFO [Node.<init>] - Binding additional locator mechanisms: relative
2024-02-29 09:41:18 07:41:18.474 INFO [GridModel.setAvailability] - Switching Node 36527a08-dc88-4d41-98d9-c3e5f2ddfd55 (uri: http://172.17.0.2:4444) from DOWN to UP
2024-02-29 09:41:18 07:41:18.474 INFO [LocalDistributor.add] - Added node 36527a08-dc88-4d41-98d9-c3e5f2ddfd55 at http://172.17.0.2:4444. Health check every 120s
2024-02-29 09:41:18 07:41:18.564 INFO [Standalone.execute] - Started Selenium Standalone 4.18.1 (revision b1d3319b48): http://172.17.0.2:4444
2024-02-29 09:41:19 2024-02-29 07:41:19,593 INFO exited: vnc (terminated by SIGTRAP; not expected)
2024-02-29 09:41:55 07:41:55.427 INFO [LocalDistributor.newSession] - Session request received by the Distributor: 
2024-02-29 09:41:55  [Capabilities {browserName: chrome}]
2024-02-29 09:41:58 07:41:58.162 INFO [LocalNode.newSession] - Session created by the Node. Id: 50d83fc47fe5ae8ee34211cef4f33abd, Caps: Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 122.0.6261.69, chrome: {chromedriverVersion: 122.0.6261.69 (81bc525b6a36..., userDataDir: /tmp/.org.chromium.Chromium...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:32893}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: linux, proxy: Proxy(), se:bidiEnabled: false, se:cdp: ws://172.17.0.2:4444/sessio..., se:cdpVersion: 122.0.6261.69, se:vnc: ws://172.17.0.2:4444/sessio..., se:vncEnabled: true, se:vncLocalAddress: ws://172.17.0.2:7900, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
2024-02-29 09:41:58 07:41:58.175 INFO [LocalDistributor.newSession] - Session created by the Distributor. Id: 50d83fc47fe5ae8ee34211cef4f33abd 
2024-02-29 09:41:58  Caps: Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 122.0.6261.69, chrome: {chromedriverVersion: 122.0.6261.69 (81bc525b6a36..., userDataDir: /tmp/.org.chromium.Chromium...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:32893}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: linux, proxy: Proxy(), se:bidiEnabled: false, se:cdp: ws://172.17.0.2:4444/sessio..., se:cdpVersion: 122.0.6261.69, se:vnc: ws://172.17.0.2:4444/sessio..., se:vncEnabled: true, se:vncLocalAddress: ws://172.17.0.2:7900, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
2024-02-29 09:42:04 07:42:04.592 INFO [ProxyNodeWebsockets.createWsEndPoint] - Establishing connection to ws://172.17.0.2:7900
2024-02-29 09:42:04 07:42:04.617 WARN [DefaultChannelPipeline.onUnhandledInboundException] - An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
2024-02-29 09:42:04 java.lang.NullPointerException
2024-02-29 09:42:04 at org.openqa.selenium.netty.server.RequestConverter.channelRead0(RequestConverter.java:127)
2024-02-29 09:42:04 at org.openqa.selenium.netty.server.RequestConverter.channelRead0(RequestConverter.java:52)
2024-02-29 09:42:04 at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2024-02-29 09:42:04 at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:102)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2024-02-29 09:42:04 at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:102)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2024-02-29 09:42:04 at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:93)
2024-02-29 09:42:04 at org.openqa.selenium.netty.server.WebSocketUpgradeHandler.channelRead(WebSocketUpgradeHandler.java:98)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2024-02-29 09:42:04 at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:93)
2024-02-29 09:42:04 at io.netty.handler.codec.http.websocketx.extensions.WebSocketServerExtensionHandler.channelRead(WebSocketServerExtensionHandler.java:91)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2024-02-29 09:42:04 at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:93)
2024-02-29 09:42:04 at io.netty.handler.codec.http.HttpServerKeepAliveHandler.channelRead(HttpServerKeepAliveHandler.java:64)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2024-02-29 09:42:04 at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436)
2024-02-29 09:42:04 at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
2024-02-29 09:42:04 at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
2024-02-29 09:42:04 at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
2024-02-29 09:42:04 at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
2024-02-29 09:42:04 at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
2024-02-29 09:42:04 at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
2024-02-29 09:42:04 at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
2024-02-29 09:42:04 at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
2024-02-29 09:42:04 at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
2024-02-29 09:42:04 at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
2024-02-29 09:42:04 at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
2024-02-29 09:42:04 at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
2024-02-29 09:42:04 at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
2024-02-29 09:42:04 at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2024-02-29 09:42:04 at java.base/java.lang.Thread.run(Thread.java:829)
2024-02-29 09:42:06 07:42:06.684 INFO [LocalSessionMap.lambda$new$0] - Deleted session from local Session Map, Id: 50d83fc47fe5ae8ee34211cef4f33abd
2024-02-29 09:42:06 07:42:06.685 INFO [GridModel.release] - Releasing slot for session id 50d83fc47fe5ae8ee34211cef4f33abd
2024-02-29 09:42:06 07:42:06.686 INFO [SessionSlot.stop] - Stopping session 50d83fc47fe5ae8ee34211cef4f33abd
```


### Operating System

macOS Sonoma 14.2.1 (23C71) Apple M3

### Docker Selenium version (image tag)

4.18.1-20240224

### Selenium Grid chart version (chart version)

_No response_

4.19.0 did not fix the issue. It also presented a new issue. node-docker does not get created successfully when trying to create this node-docker on the same instance as the selenium hub. Logs below

> Attaching to node-docker-1, selenium-hub
selenium-hub   | 2024-03-28 13:14:33,965 INFO Included extra file "/etc/supervisor/conf.d/selenium-grid-hub.conf" during parsing
selenium-hub   | 2024-03-28 13:14:33,968 INFO RPC interface 'supervisor' initialized
selenium-hub   | 2024-03-28 13:14:33,968 CRIT Server 'unix_http_server' running without any HTTP authentication checking
selenium-hub   | 2024-03-28 13:14:33,968 INFO supervisord started with pid 8
node-docker-1  | 2024-03-28 13:14:34,164 INFO Included extra file "/etc/supervisor/conf.d/selenium-grid-docker.conf" during parsing
node-docker-1  | 2024-03-28 13:14:34,167 INFO RPC interface 'supervisor' initialized
node-docker-1  | 2024-03-28 13:14:34,167 CRIT Server 'unix_http_server' running without any HTTP authentication checking
node-docker-1  | 2024-03-28 13:14:34,168 INFO supervisord started with pid 8
selenium-hub   | 2024-03-28 13:14:34,972 INFO spawned: 'selenium-grid-hub' with pid 9
selenium-hub   | Appending Selenium options: --reject-unsupported-caps false
selenium-hub   | 2024-03-28 13:14:34,978 INFO success: selenium-grid-hub entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
selenium-hub   | Tracing is disabled
node-docker-1  | 2024-03-28 13:14:35,171 INFO spawned: 'socat' with pid 9
node-docker-1  | 2024-03-28 13:14:35,173 INFO spawned: 'selenium-grid-docker' with pid 10
node-docker-1  | Starting Selenium Grid Node Docker...
node-docker-1  | Appending Selenium options: --session-timeout 297 --enable-managed-downloads true
node-docker-1  | 2024-03-28 13:14:35,178 INFO success: socat entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
node-docker-1  | 2024-03-28 13:14:35,178 INFO success: selenium-grid-docker entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
node-docker-1  | Tracing is disabled
selenium-hub   | 13:14:35.283 INFO [LoggingOptions.configureLogEncoding] - Using the system default encoding
selenium-hub   | 13:14:35.288 INFO [OpenTelemetryTracer.createTracer] - Using OpenTelemetry for tracing
selenium-hub   | 13:14:35.370 INFO [BoundZmqEventBus.<init>] - XPUB binding to [binding to tcp://*:4442, advertising as tcp://172.19.0.2:4442], XSUB binding to [binding to tcp://*:4443, advertising as tcp://172.19.0.2:4443]
selenium-hub   | 13:14:35.414 INFO [UnboundZmqEventBus.<init>] - Connecting to tcp://172.19.0.2:4442 and tcp://172.19.0.2:4443
selenium-hub   | 13:14:35.429 INFO [UnboundZmqEventBus.<init>] - Sockets created
node-docker-1  | 13:14:35.490 INFO [LoggingOptions.configureLogEncoding] - Using the system default encoding
node-docker-1  | 13:14:35.495 INFO [OpenTelemetryTracer.createTracer] - Using OpenTelemetry for tracing
node-docker-1  | 13:14:35.588 INFO [UnboundZmqEventBus.<init>] - Connecting to tcp://selenium-hub:4442 and tcp://selenium-hub:4443
node-docker-1  | 13:14:35.626 INFO [UnboundZmqEventBus.<init>] - Sockets created
selenium-hub   | 13:14:36.430 INFO [UnboundZmqEventBus.<init>] - Event bus ready
node-docker-1  | 13:14:36.628 INFO [UnboundZmqEventBus.<init>] - Event bus ready
node-docker-1  | 13:14:36.709 INFO [NodeServer.createHandlers] - Reporting self as: http://172.19.0.3:5555
node-docker-1  | 13:14:36.728 INFO [NodeOptions.getSessionFactories] - Detected 32 available processors
selenium-hub   | 13:14:36.944 INFO [Hub.execute] - Started Selenium Hub 4.19.0 (revision 5f9cec8963): http://172.19.0.2:4444
node-docker-1  | 13:14:36.997 INFO [V141Docker.isContainerPresent] - Checking if container is present: 2b314e58dc8c
node-docker-1  | 13:14:37.094 INFO [Docker.getImage] - Obtaining image: selenium/standalone-chrome
node-docker-1  | 13:14:37.094 INFO [Docker.getImage] - Obtaining image: selenium/standalone-firefox
node-docker-1  | 13:14:37.094 INFO [Docker.getImage] - Obtaining image: selenium/standalone-edge
node-docker-1  | 13:14:37.104 INFO [V141Docker.getImage] - Listing local images: Reference{domain='docker.io', name='selenium/standalone-firefox', tag='latest', digest='null'}
node-docker-1  | 13:14:37.105 INFO [V141Docker.getImage] - Listing local images: Reference{domain='docker.io', name='selenium/standalone-edge', tag='latest', digest='null'}
node-docker-1  | 13:14:37.105 INFO [V141Docker.getImage] - Listing local images: Reference{domain='docker.io', name='selenium/standalone-chrome', tag='latest', digest='null'}
node-docker-1  | 13:14:37.117 INFO [V141Docker.getImage] - Pulling Reference{domain='docker.io', name='selenium/standalone-edge', tag='latest', digest='null'}
node-docker-1  | 13:14:37.117 INFO [V141Docker.getImage] - Pulling Reference{domain='docker.io', name='selenium/standalone-firefox', tag='latest', digest='null'}
node-docker-1  | 13:14:37.117 INFO [V141Docker.getImage] - Pulling Reference{domain='docker.io', name='selenium/standalone-chrome', tag='latest', digest='null'}
node-docker-1  | 13:14:37.118 INFO [PullImage.apply] - Pulling Reference{domain='docker.io', name='selenium/standalone-firefox', tag='latest', digest='null'}
node-docker-1  | 13:14:37.118 INFO [PullImage.apply] - Pulling Reference{domain='docker.io', name='selenium/standalone-edge', tag='latest', digest='null'}
node-docker-1  | 13:14:37.118 INFO [PullImage.apply] - Pulling Reference{domain='docker.io', name='selenium/standalone-chrome', tag='latest', digest='null'}
node-docker-1  | 13:14:37.120 ERROR [Bootstrap.runMain] - Error during execution
node-docker-1  | java.lang.reflect.InvocationTargetException
node-docker-1  | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
node-docker-1  | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
node-docker-1  | 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
node-docker-1  | 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
node-docker-1  | 	at org.openqa.selenium.grid.Bootstrap.runMain(Bootstrap.java:77)
node-docker-1  | 	at org.openqa.selenium.grid.Bootstrap.main(Bootstrap.java:70)
node-docker-1  | Caused by: org.openqa.selenium.grid.config.ConfigException: java.lang.reflect.InvocationTargetException
node-docker-1  | 	at org.openqa.selenium.grid.config.MemoizedConfig.getClass(MemoizedConfig.java:119)
node-docker-1  | 	at org.openqa.selenium.grid.node.config.NodeOptions.getNode(NodeOptions.java:181)
node-docker-1  | 	at org.openqa.selenium.grid.node.httpd.NodeServer.createHandlers(NodeServer.java:126)
node-docker-1  | 	at org.openqa.selenium.grid.node.httpd.NodeServer.asServer(NodeServer.java:185)
node-docker-1  | 	at org.openqa.selenium.grid.node.httpd.NodeServer.execute(NodeServer.java:247)
node-docker-1  | 	at org.openqa.selenium.grid.TemplateGridCommand.lambda$configure$4(TemplateGridCommand.java:122)
node-docker-1  | 	at org.openqa.selenium.grid.Main.launch(Main.java:83)
node-docker-1  | 	at org.openqa.selenium.grid.Main.go(Main.java:56)
node-docker-1  | 	at org.openqa.selenium.grid.Main.main(Main.java:41)
node-docker-1  | 	... 6 more
node-docker-1  | Caused by: java.lang.reflect.InvocationTargetException
node-docker-1  | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
node-docker-1  | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
node-docker-1  | 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
node-docker-1  | 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
node-docker-1  | 	at org.openqa.selenium.grid.config.ClassCreation.callCreateMethod(ClassCreation.java:51)
node-docker-1  | 	at org.openqa.selenium.grid.config.MemoizedConfig.lambda$getClass$4(MemoizedConfig.java:104)
node-docker-1  | 	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(Unknown Source)
node-docker-1  | 	at org.openqa.selenium.grid.config.MemoizedConfig.getClass(MemoizedConfig.java:99)
node-docker-1  | 	... 14 more
node-docker-1  | Caused by: java.lang.IllegalArgumentException: non-positive contentLength: 0
node-docker-1  | 	at java.net.http/java.net.http.HttpRequest$BodyPublishers.fromPublisher(Unknown Source)
node-docker-1  | 	at org.openqa.selenium.remote.http.jdk.JdkHttpMessages.notChunkingBodyPublisher(JdkHttpMessages.java:124)
node-docker-1  | 	at org.openqa.selenium.remote.http.jdk.JdkHttpMessages.createRequest(JdkHttpMessages.java:82)
node-docker-1  | 	at org.openqa.selenium.remote.http.jdk.JdkHttpClient.execute0(JdkHttpClient.java:371)
node-docker-1  | 	at org.openqa.selenium.remote.http.AddSeleniumUserAgent.lambda$apply$0(AddSeleniumUserAgent.java:42)
node-docker-1  | 	at org.openqa.selenium.remote.http.Filter.lambda$andFinally$1(Filter.java:55)
node-docker-1  | 	at org.openqa.selenium.remote.http.jdk.JdkHttpClient.execute(JdkHttpClient.java:352)
node-docker-1  | 	at org.openqa.selenium.remote.tracing.TracedHttpClient.execute(TracedHttpClient.java:54)
node-docker-1  | 	at org.openqa.selenium.docker.v1_41.PullImage.apply(PullImage.java:62)
node-docker-1  | 	at org.openqa.selenium.docker.v1_41.V141Docker.getImage(V141Docker.java:79)
node-docker-1  | 	at org.openqa.selenium.docker.Docker.lambda$getImage$0(Docker.java:50)
node-docker-1  | 	at java.base/java.util.Optional.map(Unknown Source)
node-docker-1  | 	at org.openqa.selenium.docker.Docker.getImage(Docker.java:50)
node-docker-1  | 	at org.openqa.selenium.grid.node.docker.DockerOptions.lambda$loadImages$5(DockerOptions.java:259)
node-docker-1  | 	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
node-docker-1  | 	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(Unknown Source)
node-docker-1  | 	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
node-docker-1  | 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
node-docker-1  | 	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
node-docker-1  | 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
node-docker-1  | 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
node-docker-1  | Exception in thread "Thread-0" java.lang.NullPointerException: Cannot invoke "org.openqa.selenium.grid.node.Node.getStatus()" because "this.node" is null
node-docker-1  | 	at org.openqa.selenium.grid.node.httpd.NodeServer.lambda$new$0(NodeServer.java:78)
node-docker-1  | 	at java.base/java.lang.Thread.run(Unknown Source)
node-docker-1  | 2024-03-28 13:14:37,447 INFO exited: selenium-grid-docker (exit status 1; not expected)

@VietND96 
I am unable to test 4.19.0 because it is throwing the error below. nnode-docker does not get created successfully

```
Attaching to node-docker-1, selenium-hub
selenium-hub | 2024-03-28 13:14:33,965 INFO Included extra file "/etc/supervisor/conf.d/selenium-grid-hub.conf" during parsing
selenium-hub | 2024-03-28 13:14:33,968 INFO RPC interface 'supervisor' initialized
selenium-hub | 2024-03-28 13:14:33,968 CRIT Server 'unix_http_server' running without any HTTP authentication checking
selenium-hub | 2024-03-28 13:14:33,968 INFO supervisord started with pid 8
node-docker-1 | 2024-03-28 13:14:34,164 INFO Included extra file "/etc/supervisor/conf.d/selenium-grid-docker.conf" during parsing
node-docker-1 | 2024-03-28 13:14:34,167 INFO RPC interface 'supervisor' initialized
node-docker-1 | 2024-03-28 13:14:34,167 CRIT Server 'unix_http_server' running without any HTTP authentication checking
node-docker-1 | 2024-03-28 13:14:34,168 INFO supervisord started with pid 8
selenium-hub | 2024-03-28 13:14:34,972 INFO spawned: 'selenium-grid-hub' with pid 9
selenium-hub | Appending Selenium options: --reject-unsupported-caps false
selenium-hub | 2024-03-28 13:14:34,978 INFO success: selenium-grid-hub entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
selenium-hub | Tracing is disabled
node-docker-1 | 2024-03-28 13:14:35,171 INFO spawned: 'socat' with pid 9
node-docker-1 | 2024-03-28 13:14:35,173 INFO spawned: 'selenium-grid-docker' with pid 10
node-docker-1 | Starting Selenium Grid Node Docker...
node-docker-1 | Appending Selenium options: --session-timeout 297 --enable-managed-downloads true
node-docker-1 | 2024-03-28 13:14:35,178 INFO success: socat entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
node-docker-1 | 2024-03-28 13:14:35,178 INFO success: selenium-grid-docker entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
node-docker-1 | Tracing is disabled
selenium-hub | 13:14:35.283 INFO [LoggingOptions.configureLogEncoding] - Using the system default encoding
selenium-hub | 13:14:35.288 INFO [OpenTelemetryTracer.createTracer] - Using OpenTelemetry for tracing
selenium-hub | 13:14:35.370 INFO [BoundZmqEventBus.] - XPUB binding to [binding to tcp://:4442, advertising as tcp://172.19.0.2:4442], XSUB binding to [binding to tcp://:4443, advertising as tcp://172.19.0.2:4443]
selenium-hub | 13:14:35.414 INFO [UnboundZmqEventBus.] - Connecting to tcp://172.19.0.2:4442 and tcp://172.19.0.2:4443
selenium-hub | 13:14:35.429 INFO [UnboundZmqEventBus.] - Sockets created
node-docker-1 | 13:14:35.490 INFO [LoggingOptions.configureLogEncoding] - Using the system default encoding
node-docker-1 | 13:14:35.495 INFO [OpenTelemetryTracer.createTracer] - Using OpenTelemetry for tracing
node-docker-1 | 13:14:35.588 INFO [UnboundZmqEventBus.] - Connecting to tcp://selenium-hub:4442 and tcp://selenium-hub:4443
node-docker-1 | 13:14:35.626 INFO [UnboundZmqEventBus.] - Sockets created
selenium-hub | 13:14:36.430 INFO [UnboundZmqEventBus.] - Event bus ready
node-docker-1 | 13:14:36.628 INFO [UnboundZmqEventBus.] - Event bus ready
node-docker-1 | 13:14:36.709 INFO [NodeServer.createHandlers] - Reporting self as: http://172.19.0.3:5555
node-docker-1 | 13:14:36.728 INFO [NodeOptions.getSessionFactories] - Detected 32 available processors
selenium-hub | 13:14:36.944 INFO [Hub.execute] - Started Selenium Hub 4.19.0 (revision 5f9cec8963): http://172.19.0.2:4444
node-docker-1 | 13:14:36.997 INFO [V141Docker.isContainerPresent] - Checking if container is present: 2b314e58dc8c
node-docker-1 | 13:14:37.094 INFO [Docker.getImage] - Obtaining image: selenium/standalone-chrome
node-docker-1 | 13:14:37.094 INFO [Docker.getImage] - Obtaining image: selenium/standalone-firefox
node-docker-1 | 13:14:37.094 INFO [Docker.getImage] - Obtaining image: selenium/standalone-edge
node-docker-1 | 13:14:37.104 INFO [V141Docker.getImage] - Listing local images: Reference{domain='docker.io', name='selenium/standalone-firefox', tag='latest', digest='null'}
node-docker-1 | 13:14:37.105 INFO [V141Docker.getImage] - Listing local images: Reference{domain='docker.io', name='selenium/standalone-edge', tag='latest', digest='null'}
node-docker-1 | 13:14:37.105 INFO [V141Docker.getImage] - Listing local images: Reference{domain='docker.io', name='selenium/standalone-chrome', tag='latest', digest='null'}
node-docker-1 | 13:14:37.117 INFO [V141Docker.getImage] - Pulling Reference{domain='docker.io', name='selenium/standalone-edge', tag='latest', digest='null'}
node-docker-1 | 13:14:37.117 INFO [V141Docker.getImage] - Pulling Reference{domain='docker.io', name='selenium/standalone-firefox', tag='latest', digest='null'}
node-docker-1 | 13:14:37.117 INFO [V141Docker.getImage] - Pulling Reference{domain='docker.io', name='selenium/standalone-chrome', tag='latest', digest='null'}
node-docker-1 | 13:14:37.118 INFO [PullImage.apply] - Pulling Reference{domain='docker.io', name='selenium/standalone-firefox', tag='latest', digest='null'}
node-docker-1 | 13:14:37.118 INFO [PullImage.apply] - Pulling Reference{domain='docker.io', name='selenium/standalone-edge', tag='latest', digest='null'}
node-docker-1 | 13:14:37.118 INFO [PullImage.apply] - Pulling Reference{domain='docker.io', name='selenium/standalone-chrome', tag='latest', digest='null'}
node-docker-1 | 13:14:37.120 ERROR [Bootstrap.runMain] - Error during execution
node-docker-1 | java.lang.reflect.InvocationTargetException
node-docker-1 | at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
node-docker-1 | at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
node-docker-1 | at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
node-docker-1 | at java.base/java.lang.reflect.Method.invoke(Unknown Source)
node-docker-1 | at org.openqa.selenium.grid.Bootstrap.runMain(Bootstrap.java:77)
node-docker-1 | at org.openqa.selenium.grid.Bootstrap.main(Bootstrap.java:70)
node-docker-1 | Caused by: org.openqa.selenium.grid.config.ConfigException: java.lang.reflect.InvocationTargetException
node-docker-1 | at org.openqa.selenium.grid.config.MemoizedConfig.getClass(MemoizedConfig.java:119)
node-docker-1 | at org.openqa.selenium.grid.node.config.NodeOptions.getNode(NodeOptions.java:181)
node-docker-1 | at org.openqa.selenium.grid.node.httpd.NodeServer.createHandlers(NodeServer.java:126)
node-docker-1 | at org.openqa.selenium.grid.node.httpd.NodeServer.asServer(NodeServer.java:185)
node-docker-1 | at org.openqa.selenium.grid.node.httpd.NodeServer.execute(NodeServer.java:247)
node-docker-1 | at org.openqa.selenium.grid.TemplateGridCommand.lambda$configure$4(TemplateGridCommand.java:122)
node-docker-1 | at org.openqa.selenium.grid.Main.launch(Main.java:83)
node-docker-1 | at org.openqa.selenium.grid.Main.go(Main.java:56)
node-docker-1 | at org.openqa.selenium.grid.Main.main(Main.java:41)
node-docker-1 | ... 6 more
node-docker-1 | Caused by: java.lang.reflect.InvocationTargetException
node-docker-1 | at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
node-docker-1 | at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
node-docker-1 | at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
node-docker-1 | at java.base/java.lang.reflect.Method.invoke(Unknown Source)
node-docker-1 | at org.openqa.selenium.grid.config.ClassCreation.callCreateMethod(ClassCreation.java:51)
node-docker-1 | at org.openqa.selenium.grid.config.MemoizedConfig.lambda$getClass$4(MemoizedConfig.java:104)
node-docker-1 | at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(Unknown Source)
node-docker-1 | at org.openqa.selenium.grid.config.MemoizedConfig.getClass(MemoizedConfig.java:99)
node-docker-1 | ... 14 more
node-docker-1 | Caused by: java.lang.IllegalArgumentException: non-positive contentLength: 0
node-docker-1 | at java.net.http/java.net.http.HttpRequest$BodyPublishers.fromPublisher(Unknown Source)
node-docker-1 | at org.openqa.selenium.remote.http.jdk.JdkHttpMessages.notChunkingBodyPublisher(JdkHttpMessages.java:124)
node-docker-1 | at org.openqa.selenium.remote.http.jdk.JdkHttpMessages.createRequest(JdkHttpMessages.java:82)
node-docker-1 | at org.openqa.selenium.remote.http.jdk.JdkHttpClient.execute0(JdkHttpClient.java:371)
node-docker-1 | at org.openqa.selenium.remote.http.AddSeleniumUserAgent.lambda$apply$0(AddSeleniumUserAgent.java:42)
node-docker-1 | at org.openqa.selenium.remote.http.Filter.lambda$andFinally$1(Filter.java:55)
node-docker-1 | at org.openqa.selenium.remote.http.jdk.JdkHttpClient.execute(JdkHttpClient.java:352)
node-docker-1 | at org.openqa.selenium.remote.tracing.TracedHttpClient.execute(TracedHttpClient.java:54)
node-docker-1 | at org.openqa.selenium.docker.v1_41.PullImage.apply(PullImage.java:62)
node-docker-1 | at org.openqa.selenium.docker.v1_41.V141Docker.getImage(V141Docker.java:79)
node-docker-1 | at org.openqa.selenium.docker.Docker.lambda$getImage$0(Docker.java:50)
node-docker-1 | at java.base/java.util.Optional.map(Unknown Source)
node-docker-1 | at org.openqa.selenium.docker.Docker.getImage(Docker.java:50)
node-docker-1 | at org.openqa.selenium.grid.node.docker.DockerOptions.lambda$loadImages$5(DockerOptions.java:259)
node-docker-1 | at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
node-docker-1 | at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(Unknown Source)
node-docker-1 | at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
node-docker-1 | at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
node-docker-1 | at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
node-docker-1 | at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
node-docker-1 | at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
node-docker-1 | Exception in thread "Thread-0" java.lang.NullPointerException: Cannot invoke "org.openqa.selenium.grid.node.Node.getStatus()" because "this.node" is null
node-docker-1 | at org.openqa.selenium.grid.node.httpd.NodeServer.lambda$new$0(NodeServer.java:78)
node-docker-1 | at java.base/java.lang.Thread.run(Unknown Source)
node-docker-1 | 2024-03-28 13:14:37,447 INFO exited: selenium-grid-docker (exit status 1; not expected)
```

--- a/NodeBase/start-novnc.sh
+++ b/NodeBase/start-novnc.sh
@@ -7,12 +7,14 @@ if [ "${START_XVFB:-$SE_START_XVFB}" = true ] ; then
     if [ "${START_NO_VNC:-$SE_START_NO_VNC}" = true ] ; then
 
       # Guard against unreasonably high nofile limits. See https://github.com/SeleniumHQ/docker-selenium/issues/2045
-      ULIMIT=${SE_VNC_ULIMIT:-100000}
-      if [[ ${ULIMIT} -ge 100000 ]]; then
-        echo "Trying to update the open file descriptor limit from $(ulimit -n) to ${ULIMIT}."
-        ulimit -Sv ${ULIMIT}
+      # Try to set a new limit if the current limit is too high, or the user explicitly specified a custom limit
+      TOO_HIGH_ULIMIT=100000
+      if [[ $(ulimit -n) -gt $TOO_HIGH_ULIMIT || ! -z "${SE_VNC_ULIMIT}" ]]; then
+        NEW_ULIMIT=${SE_VNC_ULIMIT:-${TOO_HIGH_ULIMIT}}
+        echo "Trying to update the open file descriptor limit from $(ulimit -n) to ${NEW_ULIMIT}."
+        ulimit -n ${NEW_ULIMIT}
         if [ $? -eq 0 ]; then
-          echo "Successfully update the open file descriptor limit."
+          echo "Successfully updated the open file descriptor limit."
         else
           echo "The open file descriptor limit could not be updated."
         fi

--- a/NodeBase/start-vnc.sh
+++ b/NodeBase/start-vnc.sh
@@ -45,12 +45,14 @@ if [ "${START_XVFB:-$SE_START_XVFB}" = true ] ; then
     done
 
     # Guard against unreasonably high nofile limits. See https://github.com/SeleniumHQ/docker-selenium/issues/2045
-    ULIMIT=${SE_VNC_ULIMIT:-100000}
-    if [[ ${ULIMIT} -ge 100000 ]]; then
-      echo "Trying to update the open file descriptor limit from $(ulimit -n) to ${ULIMIT}."
-      ulimit -Sv ${ULIMIT}
+    # Try to set a new limit if the current limit is too high, or the user explicitly specified a custom limit
+    TOO_HIGH_ULIMIT=100000
+    if [[ $(ulimit -n) -gt $TOO_HIGH_ULIMIT || ! -z "${SE_VNC_ULIMIT}" ]]; then
+      NEW_ULIMIT=${SE_VNC_ULIMIT:-${TOO_HIGH_ULIMIT}}
+      echo "Trying to update the open file descriptor limit from $(ulimit -n) to ${NEW_ULIMIT}."
+      ulimit -n ${NEW_ULIMIT}
       if [ $? -eq 0 ]; then
-        echo "Successfully update the open file descriptor limit."
+        echo "Successfully updated the open file descriptor limit."
       else
         echo "The open file descriptor limit could not be updated."
       fi

